test/api/netcon.cpp

```diff
@@ -55,50 +47,43 @@ int main(void) {
     nrn_mechanism_insert(soma, nrn_symbol("hh"));
 
     // NetStim
-    auto ns = nrn_object_new(nrn_symbol("NetStim"), 0);
+    auto ns = nrn_object_new_NoArgs("NetStim");
     nrn_property_set(ns, "start", 5);
     nrn_property_set(ns, "noise", 1);
     nrn_property_set(ns, "interval", 5);
     nrn_property_set(ns, "number", 10);
 
     // syn = h.ExpSyn(soma(0.5))
-    nrn_double_push(0.5);
-    auto syn = nrn_object_new(nrn_symbol("ExpSyn"), 1);
+    auto syn = nrn_object_new_d("ExpSyn", 0.5);
     nrn_property_set(syn, "tau", 3);  // 3 ms timeconstant
     nrn_property_set(syn, "e", 0);    // 0 mV reversal potential (excitatory synapse)
 
     // nc = h.NetCon(ns, syn)
-    nrn_object_push(ns);
-    nrn_object_push(syn);
-    auto nc = nrn_object_new(nrn_symbol("NetCon"), 2);
+    auto nc = nrn_object_new("NetCon", "oo", ns, syn);
     nrn_property_array_set(nc, "weight", 0, 0.5);
     nrn_property_set(nc, "delay", 0);
 
-    auto vec = nrn_object_new(nrn_symbol("Vector"), 0);
-
-    // nc.record(vec)
-    nrn_object_push(vec);
-    nrn_method_call(nc, nrn_method_symbol(nc, "record"), 1);
-    // TODO: record probably put something on the stack that should be removed
+    // record from the netcon directly
+    auto vec = nrn_object_new_NoArgs("Vector");
+    nrn_method_call(nc, "record", NRN_ARG_OBJECT, vec);
 
     // setup recording
-    v = nrn_object_new(nrn_symbol("Vector"), 0);
-    nrn_rangevar_push(nrn_symbol("v"), soma, 0.5);
-    nrn_double_push(20);
-    nrn_method_call(v, nrn_method_symbol(v, "record"), 2);
```

Comment:
Looks like pushing without popping.

Reply:
With the next API we hope to avoid any manual handling of the stack. But what was there looked ok since the function would pop from the stack the number of arguments as given, in this case 2.